audb/core/dependencies.py

```diff
@@ -524,10 +525,27 @@ def _column_loc(
         files: typing.Union[str, typing.Sequence[str]],
         dtype: typing.Callable = None,
     ) -> typing.Union[typing.Any, typing.List[typing.Any]]:
-        r"""Column content for selected files."""
-        value = self._df.at[files, column]
+        r"""Column content for selected files.
+
+        Args:
+        column: one of the names in self._schema
+        files: rows to query, index is a filename
+        dtype: convert data to dtype
+
+        Returns: scalar value or list
+        """
+        # note: pandas series is not a sequence
+        is_sequence = isinstance(files, collections.abc.Sequence)
+        files = [files] if is_sequence and isinstance(files, str) else files
+
         if dtype is not None:
-            value = dtype(value)
+            value = self._df.loc[files][column].astype(dtype).tolist()
+        else:
+            value = self._df.loc[files][column].tolist()
+
+        if len(value) == 1:
+            return value[0]
```

Comment:
Unfortunately, the new implementation is too complicated and slows down the processing.

In the current `main` branch we have:

| method                                          |   string |   object |   pyarrow |
|-------------------------------------------------|----------|----------|-----------|
| Dependencies.bit_depth(10000 files)             |    0.023 |    0.023 |     0.047 |
| Dependencies.channels(10000 files)              |    0.022 |    0.024 |     0.046 |
| Dependencies.checksum(10000 files)              |    0.024 |    0.023 |     0.049 |
| Dependencies.duration(10000 files)              |    0.022 |    0.023 |     0.045 |
| Dependencies.format(10000 files)                |    0.024 |    0.022 |     0.048 |
| Dependencies.removed(10000 files)               |    0.022 |    0.022 |     0.045 |
| Dependencies.sampling_rate(10000 files)         |    0.022 |    0.022 |     0.045 |
| Dependencies.type(10000 files)                  |    0.022 |    0.022 |     0.045 |
| Dependencies.version(10000 files)               |    0.024 |    0.022 |     0.048 |

This branch results in:

| method                                          |   string |   object |   pyarrow |
|-------------------------------------------------|----------|----------|-----------|
| Dependencies.bit_depth(10000 files)             |    2.849 |    2.189 |     3.670 |
| Dependencies.channels(10000 files)              |    2.818 |    2.175 |     3.637 |
| Dependencies.checksum(10000 files)              |    2.563 |    1.875 |     3.217 |
| Dependencies.duration(10000 files)              |    2.750 |    2.112 |     3.704 |
| Dependencies.format(10000 files)                |    2.575 |    1.868 |     3.248 |
| Dependencies.removed(10000 files)               |    2.820 |    2.172 |     3.710 |
| Dependencies.sampling_rate(10000 files)         |    2.846 |    2.198 |     3.716 |
| Dependencies.type(10000 files)                  |    2.890 |    2.189 |     3.693 |
| Dependencies.version(10000 files)               |    2.597 |    1.880 |     3.237 |

Reply:
This can be tested by running:

```bash
$ python benchmark-dependencies-methods.py
```

inside the `benchmarks/` folder.

Reply:
In https://github.com/audeering/audb/pull/370, I achieved better speed for single files, by having extra code for single file vs. code for multiple files:

https://github.com/audeering/audb/blob/552108629a65313ecd77d9bfee324c2f69523ecd/audb/core/dependencies.py#L501-L522

Reply:
I see. This could be the reason then for my slow performance for single files. Then using `.at` is better for single files.